src/daemon/collector/content/collect.js

```diff
@@ -262,53 +256,52 @@ function hasSillySponsored(haystackText) {
 // THIS WORKS
 function filterFrontAds(lst) {
   let newLst = [];
-  let newStyle = false;
+  let layoutStyle;
   for (let i=0; i<lst.length; i++) {
     // detecting new FB markup style
     if (sponsoredText.indexOf(lst[i].text) > -1 && $(lst[i]).attr('href').indexOf('/ads/about/') > -1) {
-      newStyle = true;
+      layoutStyle = "FB5";
+    }
+    // this is a clickable link, not "Sponsored" link
+    const rel = lst[i].getAttribute('rel')
+    const target = lst[i].getAttribute('target')
+    if (rel || target) {
+      continue;
     }
+
     let ajaxify = lst[i].getAttribute('ajaxify');
     if (ajaxify && ajaxify.indexOf(politAdSubtitle) > -1
-      && (isScrolledIntoView(lst[i], newStyle))
+      && (isScrolledIntoView(lst[i], layoutStyle))
       && (lst[i].getAttribute('class').indexOf(non_ad) < 0)) {
         newLst.push(lst[i]);
         continue;
     }
 
-
     if (sponsoredText.indexOf(lst[i].text) > -1
       && (lst[i].getAttribute('class') && lst[i].getAttribute('class').indexOf(non_ad) < 0)
-      && isScrolledIntoView(lst[i], newStyle)) {
+      && isScrolledIntoView(lst[i], layoutStyle)) {
         newLst.push(lst[i]);
     }
-
-    if (sponsoredText.indexOf(lst[i].text) > -1
-      && (lst[i].getAttribute('class') && lst[i].getAttribute('class').indexOf(non_ad) < 0)
-      && !isScrolledIntoView(lst[i], newStyle) ){
-        // console.log(lst[i])
-        // console.log('******filter Front Ads**HIDDEN********');
-    }
-
-    if (hasSillySponsored(lst[i].text) && isScrolledIntoView(lst[i], newStyle)) {
+    // empty strings should not trigger false positives
+    if (lst[i].text && hasSillySponsored(lst[i].text) && isScrolledIntoView(lst[i], layoutStyle)) {
```

Comment:
This `hasSillySponsored` is returning too many false positives. Please can we remove this check for this release?

Reply:
ok